### DISCUSSION TITLE
Any recommendations/resources for learning the Rust APIs of Tauri?

### DISCUSSION THREAD

**Discussion by HuakunShen** [tauri-apps/tauri#7597] [Q&A]
Tauri has great documentation for JavaScript APIs https://tauri.app/, but I don't know where I can learn more about the Rust APIs. 

There doesn't seem to be much documentation or sample code on Rust.

**Comment by FabianLars** [comment]
Well, our rust api docs are hosted here: https://docs.rs/tauri/latest/tauri/index.html But they are for sure lacking. Many of them will be removed in v2 or moved into plugins where we will also work much more on the docs.

We also have a few more examples in the repo here: https://github.com/tauri-apps/tauri/tree/1.x/examples many of them focusing on rust. Also, the feature guides we currently have also (almost) all have rust code too: https://tauri.app/v1/guides/features/

But, as of Tauri v1, there aren't actually many tauri specific rust functions other than gui specific stuff like menus and dialogs, the rest is just plain rust (so check out their docs) and rust crates. This is also basically what tauri's js api, it exposes rust std stuff to the frontend.

**Reply by HuakunShen** [reply]
Thanks! 

I found it very useful to read source code from https://github.com/tauri-apps/awesome-tauri. There are lots of useful sample code in those projects.
